src/algebra/euclidean_absolute_value.lean

```diff
@@ -0,0 +1,145 @@
+/-
+Copyright (c) 2021 Anne Baanen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen
+-/
+import algebra.absolute_value
+import algebra.euclidean_domain
+import data.polynomial.field_division
+
+/-!
+# Euclidean absolute values
+
+This file defines a bundled type `euclidean_absolute_value R S` of absolute
+values compatible with the Euclidean domain structure on its domain `R`.
+
+## Main definitions
+
+ * `euclidean_absolute_value R S` is the type of absolute values on `R` mapping to `S`
+    that preserve the order on `R` arising from the Euclidean domain structure.
+ * `euclidean_absolute_value.abs` is the "standard" absolute value on `ℤ`,
+   mapping negative `x` to `-x`.
+ * `euclidean_absolute_value.card_pow_degree` is an Euclidean absolute value on
+   `𝔽_q[t]` the ring of polynomials over a finite field of cardinality `q`,
+   mapping a polynomial `p` to `q ^ degree p` (where `q ^ degree 0 = 0`)
+-/
+local infix ` ≺ `:50 := euclidean_domain.r
+
+/-- A `euclidean_absolute_value` is an absolute value `abv : R → S` that is compatible with the
+`euclidean_domain` structure on `R`, namely `abv` is strictly monotone with respect to the well
+founded relation `≺` on `R`. -/
+structure euclidean_absolute_value (R S : Type*) [euclidean_domain R] [ordered_semiring S]
+  extends absolute_value R S :=
```

Comment:
Should this `extend` or should it be a predicate on `absolute_value`s?

Reply:
It does seem easier to write out the API, at least. The real test will be to implement `absolute_value.is_admissible` in the upcoming PR :)